targets/simple_switch/primitives.cpp

```diff
@@ -59,8 +59,18 @@ class modify_field_rng_uniform
     using hash = std::hash<std::thread::id>;
     static thread_local engine generator(hash()(std::this_thread::get_id()));
     using distrib64 = std::uniform_int_distribution<uint64_t>;
-    distrib64 distribution(b.get_uint64(), e.get_uint64());
-    f.set(distribution(generator));
+    uint64_t lo = b.get_uint64();
+    uint64_t hi = e.get_uint64();
+    std::string warning = "";
+    if (lo > hi) {
+        warning = " warning: result is not specified when lo > hi";
+    }
+    distrib64 distribution(lo, hi);
+    uint64_t rand_val = distribution(generator);
+    BMLOG_TRACE_PKT(get_packet(),
```

Comment:
I'm not sure it is necessary to have such a detailed level of logging (I think we've had this discussion before), but if you want to keep it, I think the message would look better like this:
```
random(lo=..., hi=...) = ...
```